test/FastTests/Corax/Vectors/IndexVectorExactViaStaticIndexesTests.cs

```diff
@@ -0,0 +1,301 @@
+                }
+            }
+        };
+
+        await store.Maintenance.SendAsync(new PutIndexesOperation(localIndexDefinition));
+
+        var indexDefinitionFromServer = (await store.Maintenance.SendAsync(new GetIndexesOperation(0, 1))).First();
+
+
+        var cmp = localIndexDefinition.Compare(indexDefinitionFromServer);
+        Assert.Equal(IndexDefinitionCompareDifferences.None, cmp);
+    }
+
+    [RavenTheory(RavenTestCategory.Corax | RavenTestCategory.Vector)]
+    [InlineData(VectorEmbeddingType.Binary, 0.7f)]
+    [InlineData(VectorEmbeddingType.Int8, 0.82f)]
+    [InlineData(VectorEmbeddingType.Single, 0.75f)]
+    public async Task CanCreateVectorIndexFromCSharp(VectorEmbeddingType vectorEmbeddingType, float similarity)
+    {
+        using var store = CreateDocumentStore();
+        
+        {
+            using var session = store.OpenAsyncSession();
+            await session.StoreAsync(new Document() { Text = "Cat has brown eyes." });
+            await session.StoreAsync(new Document() { Text = "Apple usually is red." });
+            await session.SaveChangesAsync();
+        }
+
+        await new TextVectorIndex(vectorEmbeddingType).ExecuteAsync(store);
+        await Indexes.WaitForIndexingAsync(store);
+
+        {
+            using var session = store.OpenAsyncSession();
+            var res = session.Query<Document, TextVectorIndex>().VectorSearch(x => x.WithField(f => f.Vector), f => f.ByText("animal"), similarity);
+            var results = await res.ToListAsync();
+
+            Assert.Equal(1, results.Count);
+            Assert.Contains("Cat", results[0].Text);
+        }
+    }
+
+
+    [RavenFact(RavenTestCategory.Corax | RavenTestCategory.Vector)]
+    public async Task CreateVectorIndexFromFloatEmbeddings()
+    {
+        using var store = CreateDocumentStore();
+        {
+            using var session = store.OpenAsyncSession();
+            await session.StoreAsync(new Document() { Embeddings = [0.5f, 0.4f] });
+            await session.StoreAsync(new Document() { Embeddings = [0.1f, 0.1f] });
+            await session.StoreAsync(new Document() { Embeddings = [-0.1f, -0.1f] });
+            await session.SaveChangesAsync();
+            await new NumericalVectorIndex(VectorEmbeddingType.Single).ExecuteAsync(store);
+        }
+
+        await Indexes.WaitForIndexingAsync(store);
+        WaitForUserToContinueTheTest(store);
+    }
+
+    private IDocumentStore CreateDocumentStore() => GetDocumentStore(Options.ForSearchEngine(RavenSearchEngineMode.Corax));
+
+    private class TextVectorIndex : AbstractIndexCreationTask<Document>
+    {
+        public TextVectorIndex()
+        {
+            //querying
+        }
+
+        public TextVectorIndex(VectorEmbeddingType vectorEmbeddingType)
+        {
+            Map = docs => from doc in docs
+                select new { Id = doc.Id, Vector = CreateVectorSearch(doc.Text) };
+
+
+            VectorIndexes.Add(x => x.Vector,
```

Comment:
Why did we choose `VectorIndexes` name? Maybe just `Vectors` or `SearchVectors` (to distinguish from `TermsVectors` that we already have)?

Reply:
I actually copied the naming from spatial fields for API naming consistency.